Newsletters: add 198 (2022-05-04) (#753)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2022-05-04-newsletter.md

name: 2022-05-04-newsletter
slug: 2022-05-04-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME:harding

## News

- **MuSig2 implementation notes:** Olaoluwa Osuntokun
  [replied][osuntokun musig2] to the draft BIP for [MuSig2][topic musig]
  mentioned in [Newsletter #195][news195 musig2] with notes from the
  implementations he and others have worked on for BTCD and LND:

    - *Interaction with BIP86:* keys created by a [BIP32 HD
      wallet][topic bip32] implementing [BIP86][] follow the [BIP341][]
      recommendation for creating keypath-only keys by tweaking the key
      by a hash of itself.  This helps prevent the key from being used
      in a [multisignature][topic multisignature] which could allow one
      participant to secretly include a scriptpath spending option they
      control, giving them the ability to steal all the funds.  However,
      if the multisignature participants deliberately want to include a
      scriptpath spending option, they need to share the un-tweaked
      versions of their keys with each other.

        Osuntokun recommends that BIP86 implementations return both the

>         original key ("pre-tweaked key") and the tweaked key so that the

Author: Nice!  Updating to use those terms here.  Whenever you finish, I'd love to have a PR to the style guide with the terms and possibly links to your overview.

>         original key ("pre-tweaked key") and the tweaked key so that the

Reviewer: I have you firmly on my list of people to get feedback from!

        calling function can use whichever one is appropriate for its
        context.

    - *Interaction with scriptpath spends:* keys meant to be used with
      scriptpath spends have a related problem: in order to use the
      scriptpath, the spender must know the pre-tweaked key.  Again,
      this suggests that implementations return the pre-tweaked key so
      that it's available to be used in other code that needs it.

    - *Shortcut for final signer:* Osuntokun also sought clarification
      on a section in the BIP which describes how the final signer (and
      only the final signer) can use deterministic randomness or a
      lower-quality source of randomness for generating their signature

>       nonce.  Brandon Black [replied]black musig2] to describe the

Reviewer: ```suggestion
      nonce.  Brandon Black [replied][black musig2] to describe the
```

  signal favor] certain parts of the Bitcoin economy over others, or
  [penalizing][ivgi signal hodl voting] early voters over those who
  waited to participate in consensus formation.

    As on previous occasions where this topic has been discussed, it
    did not appear that any of the suggested methods would produce a
    result that would be sufficiently respected by most of the
    discussion participants to strongly inform their decisions about
    changing Bitcoin's consensus rules.

- **LN anchor outputs security issue:** Bastien Teinturier
  [posted][teinturier security] to the Lightning-Dev mailing list the
  announcement of a security issue he had previously responsibly
  disclosed to LN implementation maintainers.  The issue affected older
  versions of Core Lightning (with experimental features enabled) and
  LND.

    Prior to the implementation of [anchor outputs][topic anchor
    outputs], revoked [HTLC][topic HTLC] transactions only contained a
    single output, so many implementations only tried to claim that
    single output.  The design of anchor outputs for LN allows combining
    multiple revoked HTLC outputs into a single transaction, but this

>     is only safe if implementations claim all of the relevant outputs in

Reviewer: If I understand this right, the issue was that some implementations would not claim all funds that belonged to them. If so, I think the "is only safe" put me on the wrong track here. Perhaps this could be rephrased to clarify—especially if I'm still on the wrong track. :)

Author: Updating to make it clearer that it's a theft issue, thanks for letting me know about the ambiguity!

    the transaction.  Teinturier's implementation of anchor outputs for
    Eclair allowed him to test the other LN implementations and discover
    the vulnerability.

    As with a previous attack related to anchor outputs (see [Newsletter
    #115][news115 fee stealing]), the problem appears to be related to
    adding support for signing with
    `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` while still retaining legacy
    support for signing with `SIGHASH_ALL`.

- **LN gossip rate limiting:** Alex Myers [posted][myers recon] to the
  Lightning-Dev mailing list about his research into using
  [minisketch][topic minisketch]-based set reconciliation to reduce the

>   amount of bandwidth nodes need to learn about updates to the LN

Reviewer 2: ```suggestion
  amount of bandwidth nodes needed to learn about updates to the LN
```

  channel graph.  His method assumes all peers have almost all of the
  same information about almost all of the same public channels.  One
  peer can then generate a minisketch from its complete graph of the
  public network and send that to all of its peers, who can use the
  minisketch to find any updates to the network since its last
  reconciliation.  This is different than the proposed use of minisketch
  for the Bitcoin P2P network via the [erlay][topic erlay] protocol
  where only updates (new unconfirmed transactions) from the last few
  seconds are sent.

    One challenge of reconciling over all public channels is that it

>     requires all LN nodes keep the same information.  Any filtering that

Reviewer 2: ```suggestion
    requires all LN nodes to keep the same information.  Any filtering that
```

Author: I think this is better without the preposition, but let me know if there's some rule I'm missing and I'll do better in the future.

    produces a persistent difference in the view of the channel graph
    between nodes will result in either a bandwidth overhead or a
    failure of the protocol.  Matt Corallo [suggested][corallo recon]
    that this could be addressed by applying the erlay model to LN---if
    only new information was synced, there wouldn't be a concern about
    persistent differences, although large variations in filtering rules
    could still result in wasted bandwidth or reconciliation failure.
    Myers was concerned about the amount of state tracking required by

>     only sending updates---a Bitcoin Core node maintains separate state
>     for each of its peers in order to avoiding resending them updates
>     they previously sent to the node.  The alternative of reconciling

Reviewer 2: ```suggestion
    only sending updates---a Bitcoin Core node maintains a separate state
    for each of its peers in order to avoid resending updates
    they previously sent to the node.  The alternative of reconciling
```

release candidates.*

FIXME:harding to update Tuesday

- BTCPay Server 1.5.0
- BDK 0.18.0

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Bitcoin Core #18554][] wallet: ensure wallet files are not reused across chains FIXME:harding

- [Bitcoin Core #24322][] is part of a larger effort to extract out a consensus engine by creating a
  library to use Bitcoin Core's consensus code as-is, then incrementally pruning modules to make the
  library more minimal.  Namely, this PR introduces a `libbitcoinkernel` library delineating all of
  the source files the `bitcoin-chainstate` executable (introduced in [Bitcoin Core #24304][]) needs
  to link against. The list includes files that may not seem logically related to consensus,

>   illustrating the current, dependencies of Bitcoin Core's consensus engine. Future work will

Reviewer 2: ```suggestion
  illustrating the current dependencies of Bitcoin Core's consensus engine. Future work will
```
